packages/shared/src/components/fields/Dropdown.tsx

```diff
@@ -76,6 +79,10 @@ export function Dropdown({
   const triggerRef = useRef<HTMLButtonElement>();
   const { show, hideAll } = useContextMenu({ id });
 
+  useEffect(() => {
+    onOpenChange?.(isVisible);
```

Comment:
Why did our dropdown need this exactly?

Reply:
The experience level dropdown needs to know if the underlying dropdown is open or not - we change the styles depending on that.

Reply:
👍 Hmm, just thinking, not to block this PR, but maybe we can check with design if this is intentional, if we want these styles changed in every dropdown we have when it's open or not. As it seems like a new addition on their side

Reply:
👍 Hmm, just thinking, not to block this PR, but maybe we can check with design if this is intentional, if we want these styles changed in every dropdown we have when it's open or not. As it seems like a new addition on their side

Reply:
I looked at the code again and now understand why I did that. The error hint implementation is different than in Figma, to make it aligned to all the other fields (instead of showing a rounded rectangle with an exclamation mark triangle icon, we just show red text below and add red box-shadow). But it does not make sense to add the red box-shadow when the dropdown is open, hence the following code:
```
              !open &&
              !valid &&
              '!shadow-[inset_0.125rem_0_0_var(--status-error)]',
```

I'll verify with design if this is OK.

Reply:
https://dailydotdev.slack.com/archives/C06UBS0N5FU/p1713941141715529

Reply:
Do u think this should be a behaviour for only this dropdown or all of them if the field is required and not filled? I feel like this should be part of the component itself no?

Reply:
@nensidosari design is OK with the changes and will apply it to the guidelines: https://dailydotdev.slack.com/archives/C06UBS0N5FU/p1713943722338799?thread_ts=1713941141.715529&cid=C06UBS0N5FU

Reply:
> Do u think this should be a behaviour for only this dropdown or all of them if the field is required and not filled? I feel like this should be part of the component itself no?

please read this one more time, not much to do with design in general but component itself on how we implement, do we keep error states (css) within the component or outside as u're doing atm?

Reply:
As discussed I'm good if we keep this implementation and work on a refactor later on a different PR

Reply:
Created https://dailydotdev.atlassian.net/browse/MI-331 as a followup for the refactor. If I have time in the sprint, I'll try to squeeze it in :)